File: TP 1/java/palire.java
import java.util.Scanner;

class palire{


    static boolean comp(String pala,int y,int i){
        
        boolean resp=true;
        if(i<pala.length()/2){
            if(pala.charAt(i) == pala.charAt(y)){
               
            }
            else{
                resp=false;
                i=pala.length();
            }
            
            comp(pala,y-1,i+1);
        }
        return(resp);
    }
    
    
    public static void main(String[] args){
    
        Scanner sc= new Scanner(System.in);
        String pala = new String();
        pala = sc.nextLine();
            
        while(!pala.equals("FIM")){
            int y=pala.length()-1;
        int i=0;
            if(comp(pala,y,i)){
                System.out.println("SIM");
            }
            else{
                System.out.println("NAO");
            }
            pala = sc.nextLine();    
        }
        sc.close();
    }
}

File: TP 1/c/palire.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int tamanho(const char *palavra){
    int length = 0;
    
    while (palavra[length] != '\0') {
        length++;
    }

    return length;
}


bool fim(char *palavra){
    bool resp=false;
    if(tamanho(palavra) >= 3 && palavra[0] == 'F' && palavra[1] == 'I' && palavra[2] == 'M'){
        resp=true;
    }
    return(resp);
}
bool comp(char pala[], int i, int y) {
    bool resp = true;

    if (i < y) {
        if (pala[i] == pala[y]) {
            resp = comp(pala, i + 1, y - 1);  
        } else {
            resp = false;  
        }
    }
    return resp;
}

void main() {
    bool resp = 0;
    char pala[500];

    scanf("%[^\n]", pala);
    getchar();
    while (!fim(pala)) {
        int tam = tamanho(pala);
        resp = comp(pala, 0, tam - 1);

        if (resp) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }
        
        scanf("%[^\n]", pala);
        getchar();
    }
}
